### DISCUSSION TITLE
local signcolumn setting

### DISCUSSION THREAD

**Discussion by doongjohn** [nvim-neo-tree/neo-tree.nvim#104] [Q&A] [answered]
I'm using `vim.o.signcolumn = 'yes:3'` for my neovim config.\
I want to set `signcolumn=auto` only for the neo-tree window.

I've tried `event_handlers`
```lua
require'neo-tree'.setup {
  event_handlers = {
    {
      event = "vim_win_enter",
      handler = function(arg)
        vim.cmd [[setlocal scl=auto]]
      end
    }
  }
}
```
but this will set  `signcolumn=auto` for all windows 😢

**Comment by cseickel** [comment]
Yes, the vim_win_enter is just using the vim autocmd event directly and fires for all events once subscribed. You could use that and add a check for the filetype, or do the same in a generic vim BufEnter event. Either way the key is to check the filetype first:

```lua

require'neo-tree'.setup {
  event_handlers = {
    {
      event = "vim_win_enter",
      handler = function(arg)
        if vim.bo.filetype == "neo-tree" then
          vim.cmd [[setlocal scl=auto]]
        end
      end
    }
  }
}
```

**Comment by doongjohn** [comment]
Thanks this worked perfectly!
```lua
require'neo-tree'.setup {
  event_handlers = {
    {
      event = "vim_buffer_enter",
      handler = function(arg)
        if vim.bo.filetype == "neo-tree" then
          vim.cmd [[setlocal scl=auto]]
        end
      end
    }
  }
}
```

**Reply by bennypowers** [reply]
Thanks I adopted this to turn off other settings

```lua
  event_handlers = {
    {
      event = "vim_buffer_enter",
      handler = function(arg)
        if vim.bo.filetype == "neo-tree" then
          vim.cmd [[
            setlocal nocursorcolumn
            setlocal virtualedit=all
          ]]
        end
      end
    }
  },


```

**Comment by cseickel** [comment]
Glad to hear it worked out! I'm going to convert this to a discussion for others to find.

**Comment by cseickel** [comment]
For anyone else interested in this, please see this [comment](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/155#issuecomment-1052139713) on #155 and let me know your thoughts.

**Comment by rockyzhang24** [comment, marked as answer]
I think we can use `neo_tree_buffer_enter` event instead of the general `vim_buffer_enter` event. 
```lua
  event_handlers = {
    {
      event = "neo_tree_buffer_enter",
      handler = function(arg)
        vim.cmd [[
          setlocal signcolumn=no
        ]]
      end,
    }
  },
```

**Reply by cseickel** [reply]
This is correct. When the question was originally asked, we did not have a `"neo_tree_buffer_enter"` yet. This new event makes it a little easier and might handle some more edge cases now or in the future.

**Reply by rockyzhang24** [reply]
Okay I see. Thank you.
